python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -859,15 +877,16 @@ def date_range(
             # end == start, return exactly 1 timestamp (start)
             periods = 1
 
-    # The estimated upper bound of `end` is enforced to be computed to make
-    # sure overflow components are raised before actually computing the
-    # sequence.
+    # We compute `end_estim` (the estimated upper bound of the date
+    # range) below, but don't always use it.  We do this to ensure
+    # that the appropriate OverflowError is raised by Pandas in case
+    # of overflow.
     # FIXME: when `end_estim` is out of bound, but the actual `end` is not,
     # we shouldn't raise but compute the sequence as is. The trailing overflow
     # part should get trimmed at the end.
     end_estim = (
         pd.Timestamp(start.value)
-        + (periods - 1) * offset._maybe_as_fast_pandas_offset()
+        + periods * offset._maybe_as_fast_pandas_offset()
```

Comment:
Was this a bug?

Reply:
Yes, the changes below that replaces `linspace` with `arrange` also have to do with this.
See: https://github.com/rapidsai/cudf/pull/9178#discussion_r747815516

Reply:
Thanks for the context @isVoid!